### PR TITLE
Adding some more readme

### PR DESCRIPTION
**Pull request by arnokoehler**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,53 @@
...
+- validation
+- spring-data-jpa <em>(optional)</em> 
+- a database driver of your choice <em>(optional)</em> 
+- testcontainers <em>(optional)</em> 
+
+Once you have set up your project, you can start using spring-boot-crud by extending the provided base types in your data models.
+
+## Adding to your Application
+To get started with spring-boot-crud, you will want to begin by creating a controller that extends CRUDController. 
+You may choose to use one of the variants of CRUDController such as CRUDControllerSimple, CRUDControllerSimpler, or CRUDControllerSimplest depending on your needs.
```
Comment (makiftutuncu) [inline README.md:43]: Class names weren't correct so following might be better.
```suggestion
If you need all the flexibility `CRUDController` provides, you may choose to use one of the following variants `SimpleController`, `SimplerController`, or `SimplestController`.
```

File: README.md [hunk 2]
```diff
@@ -27,4 +56,134 @@ See
...
 * [CRUDEntity](api/src/main/java/dev/akif/crud/CRUDEntity.java)
 
-and their `Simple`, `Simpler` and `Simplest` variants.
+and their Simple, Simpler, and Simplest variants.
+
+## Code Examples
+
+### Writing a controller
+
+The code snippet represents a Java Controller that implements the CRUDController interface.
```
Comment (makiftutuncu) [inline README.md:65]: `CRUDController` is an abstract class so this statement and following example would need to be updated.

File: README.md [hunk 3]
```diff
@@ -27,4 +56,134 @@ See
...
+- `I`: the type of the ID of the entities
+- `E`: the type of the Entity objects
+- `M`: the type of the Model objects
+- `D`: the type of the Data Transfer Object (DTO) objects
+- `CM`: the type of the Create Model objects
+- `UM`: the type of the Update Model objects
+- `CD`: the type of the Create DTO objects
+- `UD`: the type of the Update DTO objects
+- `Mapper`: the type of the mapper class used to map between Entity and Model objects
+- `DTOMappe`r: the type of the mapper class used to map between DTO and Model objects
```
Comment (makiftutuncu) [inline README.md:104]: Small typo
```suggestion
- `DTOMapper`: the type of the mapper class used to map between DTO and Model objects
```

File: README.md [hunk 4]
```diff
@@ -27,4 +56,134 @@ See
...
+- `S`: the type of the CRUDService implementation
+
+By using these generics, the CRUDController ensures that the correct types are used for each layer of the application, and it ensures that everything works together seamlessly.
+
+To use the CRUDController in your application, simply extend it with your specific types and provide implementations for the abstract methods.
+
+```java
+@Service
+public class ExampleService implements CRUDService<
```
Comment (makiftutuncu) [inline README.md:113]: `CRUDService` is an abstract class so this statement and following example would need to be updated.

File: README.md [hunk 5]
```diff
@@ -27,4 +56,134 @@ See
...
+- `UD`: the type of the Update DTO objects
+- `Mapper`: the type of the mapper class used to map between Entity and Model objects
+- `DTOMappe`r: the type of the mapper class used to map between DTO and Model objects
+- `S`: the type of the CRUDService implementation
+
+By using these generics, the CRUDController ensures that the correct types are used for each layer of the application, and it ensures that everything works together seamlessly.
+
+To use the CRUDController in your application, simply extend it with your specific types and provide implementations for the abstract methods.
+
+```java
```
Comment (makiftutuncu) [inline README.md:111]: Is this example is here to show roughly what `CRUDService` provides?

If the answer is yes, this is fine but we should probably mention the fact that this is not the expected user code, it is here to give you an idea of what the library is doing.

If the answer is no and this is an example of the user code, then we have some redundancy. User code could (and should) be as simple as following
```java
@Service
public class ExampleService extends CRUDService<
                Long, 
                ExampleModel, 
                ExampleEntity, 
                ExampleCreateModel,
                ExampleUpdateModel, 
                ExampleMapper> {
    public ExampleService(Clock clock, ExampleRepository repository, ExampleMapper mapper) {
        super("Example", clock, repository, mapper);
    }
}
```

The same question applies to the controller example above.

File: README.md [hunk 6]
```diff
@@ -27,4 +56,134 @@ See
...
+
+By implementing this interface, the class provides a base type for CRUD operations, and it is responsible for the implementation of basic CRUD operations such as create, get, update, and delete. The implementation details will be different based on the actual requirements, but it provides a well-defined structure that makes it easier to implement CRUD operations in a consistent manner. The methods in the service class will be used in the controller layer to handle CRUD operations and return responses to clients.
+
+## API Endpoints
+
+The following API endpoints will be made available:
+
+| Method  | Endpoint  | Description  |
+|---|---|---|
+| GET	 | /api/v1/examples	        | Retrieve a list of examples |
```
Comment (makiftutuncu) [inline README.md:180]: This table is great but controller example above would need to have `@RequestMapping("api/v1/examples")`.

**Review by makiftutuncu — CHANGES_REQUESTED** [review]
Awesome work! I made some comments, mostly about the correctness of the examples. Otherwise, it's a great start to user documentation. 🥳

**Comment by makiftutuncu** [comment]
FYI, I just merged https://github.com/makiftutuncu/spring-boot-crud/pull/3 so links to files would also need to be updated.

**Comment by arnokoehler** [comment]
I'll see what I can do, today or tomorrow.

**Review by makiftutuncu — APPROVED** [review]

**Pull request merged**